src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,105 @@ private void addListingToBusinesses(PlacesSearchResult place) {
+                                          .location(latLng)
+                                          .radius(50000);
+      try {                                       
+        PlacesSearchResult[] similarBusinessesInTheArea = request
+                                                            .await()
+                                                            .results;
+        if (similarBusinessesInTheArea.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName(), similarBusinessesInTheArea);
+        }
+      } catch(GeneralSecurityException | IOException | InterruptedException | ApiException e ) {
+          LOGGER.warning(e.getMessage());
+        }                                                   
+    }
+  }
+  
+  private void checkBusinessThroughLinkedin(String currentBusinessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) 
+                              throws GeneralSecurityException, IOException {
+    String cx = "REDACTED"; 
+    Customsearch cs = new Customsearch.Builder(
+        GoogleNetHttpTransport.newTrustedTransport(), 
+        JacksonFactory.getDefaultInstance(), null) 
+              .setApplicationName("linkedinSearch") 
+              .setGoogleClientRequestInitializer(new CustomsearchRequestInitializer(KEY)) 
+              .build();
+
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    int companyFollowers = 0;
+    if (searchJsonResults!=null && searchJsonResults.size() != 0) {
+      Result linkedinBusiness = searchJsonResults.get(0);
+      String businessDescription = 
+        (String) linkedinBusiness.getPagemap().get("metatags").get(0).get("og:description");
+      if(businessDescription.indexOf(START_SUBSTRING) != -1 
+          && businessDescription.indexOf(END_SUBSTRING) != -1){
+        String followers = businessDescription.substring(
+                                businessDescription.indexOf(START_SUBSTRING) + 2, 
+                                businessDescription.indexOf(END_SUBSTRING) - 1);
+        try{
+          companyFollowers = 
+                    Integer.parseInt(followers.replaceAll(",", ""));
+        } catch (NumberFormatException e){
+            LOGGER.warning(e.getMessage());
+        }
+      }
+      if (companyFollowers > MINFOLLOWERS) {
+        addBigBusinessToDatabase();
+      } else {
+        checkNumberOfSimilarBusinessesInTheArea(currentBusiness.getName(),
+                                                similarBusinessesInTheArea);
+      }
+    }
+  }
+
+  private void checkNumberOfSimilarBusinessesInTheArea(String businessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) {
+    int countNumberOfMatchingBusiness = 0;
+    int i = 0;
+    while (i < similarBusinessesInTheArea.length 
+          && countNumberOfMatchingBusiness < 10) {
```

Comment:
Please make 10 a constant, and you might want to decrease it. How was it settled on?

Reply:
Got it

Reply:
I changed the number to 5, as it should hopefully get all the big businesses in the area